### PR TITLE
Auto YAHP

### PR DESCRIPTION
**Pull request by jbloxham**
WIP

current TODO as of `addd4f6`:
- docs parsing

### CODE REVIEW

File: tests/autoyahp/test_ppv.py [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+    def __init__(
+        self,
+        int_field: int,
+        float_field: float,
+    ):
+        self.int_field = int_field
+        self.float_field = float_field
+
+
+@hp.register_subclass(InnerChoiceClass, 'option_two')
```
Comment (ravi-mosaicml) [inline tests/autoyahp/test_ppv.py:38]: Could the register subclass decorator be replaced with `__init_subclass__` on the parent class (and the name be the `class.__name__`)

File: tests/autoyahp/test_ppv.py [hunk 2]
```diff
@@ -0,0 +1,167 @@
+import pathlib
+from typing import List
+
+import pytest
+
+import yahp as hp
+from tests.yahp_fixtures import YamlInput, generate_named_tuple_from_data
+
+
+@hp.mark_parent_provided_value('int_field')
```
Comment (ravi-mosaicml) [inline tests/autoyahp/test_ppv.py:10]: Could this be a class level attribute? E.g.

```python3 

class InnerClass:
    __yahp_ignore_fields__ = ["int_field"]
```

File: tests/autoyahp/test_ppv.py [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+
+
+@hp.create_subclass_registry()
+@hp.mark_parent_provided_value('int_field')
+class InnerChoiceClass:
+    pass
+
+
+@hp.register_subclass(InnerChoiceClass, 'option_one')
+class OptionOneClass:
```
Comment (ravi-mosaicml) [inline tests/autoyahp/test_ppv.py:27]:
```suggestion
class OptionOneClass(InnerChoiceClass):
```

**Review by ravi-mosaicml — COMMENTED** [review]
I'll continue to look tmrw! From a quick glance, impressed with how clean this could make things. Just left a few thoughts for now...

**Comment by Averylamp** [comment]
removing myself from review until the PR is ready

**Comment by Averylamp** [comment]
Closing until we decide to revisit redo this

**Pull request closed without merging**